src/i18n/provider.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { ReactNode, useEffect } from "react";
+import { I18nextProvider } from "react-i18next";
+
+import { useAuth } from "@reearth/auth";
+import { useLanguageQuery } from "@reearth/gql";
+
+import i18n from "./i18n";
+import { Provider as LegacyProvider } from "./legacy";
+
+export default function Provider({ children }: { children?: ReactNode }) {
+  const { isAuthenticated } = useAuth();
+  const { data } = useLanguageQuery({ skip: !isAuthenticated });
+  const locale = data?.me?.lang;
+  useEffect(() => {
+    i18n.changeLanguage(locale);
```

Comment:
`locale`
<img width="456" alt="Screen Shot 2022-04-27 at 11 42 20" src="https://user-images.githubusercontent.com/34051327/165429252-3dade263-8f49-44eb-b9e1-c666833f6c33.png">
, if not chosen by the user yet, apparently can be `und` which is used for the `Auto` setting.

Reply:
I tried to set undefined as the language (i.e. automatic) when und. But it needs to be tested.

Reply:
tested and it works